dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContentletLoader.java

```diff
@@ -0,0 +1,969 @@
+            .append(content.getContentTypeId())
+            .append("\" )");
+        if (type.baseType() == BaseContentType.WIDGET) {
+            sb.append("#set( $isWidget= \"")
+                .append(true)
+                .append("\")");
+            if (type.name()
+                .equals(FormAPI.FORM_WIDGET_STRUCTURE_NAME_FIELD_NAME)) {
+                sb.append("#set($isFormWidget= \"")
+                    .append(true)
+                    .append("\")");
+            } else {
+                sb.append("#set($isFormWidget= \"")
+                    .append(false)
+                    .append("\")");
+            }
+        } else {
+            sb.append("#set($isWidget= \"")
+                .append(false)
+                .append("\")");
+        }
+
+        // close contentlet div
+        sb.append("#if($API_EDIT_MODE)")
+            .append("</div>")
+            .append("#end");
+
+        writeOutVelocity(filePath, sb.toString());
+
+
+        try {
+            result = new ByteArrayInputStream(sb.toString()
+                .getBytes("UTF-8"));
+        } catch (UnsupportedEncodingException e1) {
+            result = new ByteArrayInputStream(sb.toString()
+                .getBytes());
+            Logger.error(ContainerServices.class, e1.getMessage(), e1);
+        }
+        return result;
+    }
+
+    @SuppressWarnings("unchecked")
+    public List<Contentlet> sortContentlets(List<Contentlet> contentletList, String sortBy) {
+
+        Logger.debug(this.getClass(), "I'm ordering by " + sortBy);
+
+        if (sortBy.equals("tree_order")) {
+            Collections.sort(contentletList, new WebAssetSortOrderComparator());
+        } else {
+            Collections.sort(contentletList, new ContentComparator("asc"));
+        }
+
+        return contentletList;
+
+    }
+
+    /**
+     * Will remove all contentlet files within a structure for both live and working. Uses the
+     * system user.
+     * 
+     * @param contentlets
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    public void removeContentletFile(Structure structure) throws DotDataException, DotSecurityException {
+        ContentletAPI conAPI = APILocator.getContentletAPI();
+        int limit = 500;
+        int offset = 0;
+        List<Contentlet> contentlets = conAPI.findByStructure(structure, APILocator.getUserAPI()
+            .getSystemUser(), false, limit, offset);
+        int size = contentlets.size();
+        while (size > 0) {
+            for (Contentlet contentlet : contentlets) {
+                invalidateAll(contentlet);
+            }
+            offset += limit;
+            contentlets = conAPI.findByStructure(structure, APILocator.getUserAPI()
+                .getSystemUser(), false, limit, offset);
+            size = contentlets.size();
+        }
+    }
+
+    private void removeContentletFile(Contentlet asset, Identifier identifier, boolean EDIT_MODE)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00117)